trivup/apps/SchemaRegistryApp.py

```diff
@@ -28,96 +28,115 @@
-            self.conf['extport'] = port
-            self.conf['intport'] = 8081
-            docker_args = '-p %d:%d' % (self.conf['extport'],
-                                        self.conf['intport'])
-
-        # This is the listener address inside the docker container
-        self.conf['listeners'] = 'http://0.0.0.0:%d' % self.conf.get('intport')
-        # This is the listener address outside the docker container,
-        # using port-forwarding
-        self.conf['url'] = 'http://localhost:%d' % self.conf['extport']
-
-        # Run in foreground.
-        self.conf['start_cmd'] = 'docker run %s --name %s -e SCHEMA_REGISTRY_KAFKASTORE_BOOTSTRAP_SERVERS=%s  -e SCHEMA_REGISTRY_HOST_NAME=localhost   -e SCHEMA_REGISTRY_LISTENERS=%s  -e SCHEMA_REGISTRY_DEBUG=true %s' % (  # noqa: E501
-            docker_args,
-            self.conf.get('container_id'),
-            bootstrap_servers,
-            self.conf.get('listeners'),
-            self.conf.get('image'))
-
-        # Stop through docker
-        self.conf['stop_cmd'] = 'docker stop %s' % \
-                                self.conf.get('container_id')
+        ssl = cluster.find_app(SslApp)
+        if ssl is not None:
+            # Mount SslApp dir, reuse broker trusts/certs
+            self.add_mount(self.docker_args, "/opt/ssl", ssl.root_path())
+            protocol = 'https'
+            app_conf.update({
+                'listeners': 'https://0.0.0.0:8081',
+                'inter_instance_protocol': 'https',
+                'ssl_keystore_location':
+                    '/opt/ssl/broker{}.keystore.jks'.format(kafka.appid),
+                'ssl_keystore_password': ssl.conf.get('ssl_key_pass'),
+                'ssl_key_password': ssl.conf.get('ssl_key_pass'),
+                'ssl_truststore_location':
+                    '/opt/ssl/broker{}.truststore.jks'.format(kafka.appid),
+                'ssl_truststore_password':  ssl.conf.get('ssl_key_pass')
+                })
+
+        if bool(self.conf.get('enable_auth', False)):
+            self.conf['userinfo'] = 'ckp_tester:test_secret'
+            app_conf.update({'authentication.method': 'BASIC',
+                             'authentication.realm': 'SchemaRegistry',
+                             'authentication.roles': 'Testers'})
+            app_conf['OPTS'] =\
+                '-Djava.security.auth.login.config=/opt/app/'\
+                'schema-registry.jaas'
+
+        if bool(self.conf.get('enable_auth', False)):
+            self.conf['userinfo'] = 'ckp_tester:test_secret'
+            self.add_mount(self.docker_args, "/opt/app", self.root_path())
+            self.create_file_from_template('login.properties')
+            self.create_file_from_template('schema-registry.jaas', subst=False)
+
+        self.conf['url'] = '{}://{}@localhost:{}'.format(
+                protocol,
+                self.conf.get('userinfo', ''),
+                port)
+
+        # format configs for use by docker
```

Comment:
docker itself does not care about these envs, maybe 'the schema-registry docker image'